### PR TITLE
Android favorite

### LINKED ISSUE #326: Android favorite/unfavorite feature
#### Issue description
- User should be able to favorite an item he/she is currently viewing. There is already a favorite icon in the ActionBar. Request functionality should be implemented.

### PR DESCRIPTION
**Pull request by mmutas**
Fixes #326  .

Changes proposed in this pull request:
- Favorite feature added to android.
-- There is number of favorites of each item on the feed.
-- You can add or delete an item from your favorites.

### CODE REVIEW

File: Atlas/android/app/src/main/java/com/bounswe2017/group10/atlas/httpbody/CultureItem.java [hunk 1]
```diff
@@ -66,6 +66,14 @@
     @Expose
     private boolean publicAccessibility;
 
+    @SerializedName("is_favorite")
```
Comment (eozd) [inline Atlas/android/app/src/main/java/com/bounswe2017/group10/atlas/httpbody/CultureItem.java:69]: When you add a new field to CultureItem, you have to change the Parcelable implementation to parcel this field as well. To do this for ```isFavorite```,
1. you need to read a boolean into ```isFavorite``` in ```CultureItem(Parcel in)``` constructor,
2. you need to write a boolean value into ```out``` Parcel in ```writeToParcel``` function.
3. The order in which you read from parcel, and you write to parcel must be the same, i.e. if you read into ```isFavorite``` after reading into ```publicAccessibility```, you need to write into ```isFavorite```  after you write into ```publicAccessibility```.

File: Atlas/android/app/src/main/java/com/bounswe2017/group10/atlas/httpbody/CultureItem.java [hunk 2]
```diff
@@ -66,6 +66,14 @@
...
+    @Expose
+    private boolean isFavorite;
+
+    @SerializedName("favorited_amount")
```
Comment (eozd) [inline Atlas/android/app/src/main/java/com/bounswe2017/group10/atlas/httpbody/CultureItem.java:73]: Do the same steps for Parcelable implementation as you will do for ```isFavorite```.

**Review by eozd — CHANGES_REQUESTED** [review]

File: Atlas/android/app/src/main/java/com/bounswe2017/group10/atlas/httpbody/CultureItem.java (hunk 1 above)
Comment (mmutas) [reply Atlas/android/app/src/main/java/com/bounswe2017/group10/atlas/httpbody/CultureItem.java:69]: done @eozd

File: Atlas/android/app/src/main/java/com/bounswe2017/group10/atlas/httpbody/CultureItem.java (hunk 2 above)
Comment (mmutas) [reply Atlas/android/app/src/main/java/com/bounswe2017/group10/atlas/httpbody/CultureItem.java:73]: done @eozd

**Review by eozd — APPROVED** [review]

**Pull request merged**